.github/workflows/formatapply.yml

```diff
@@ -0,0 +1,23 @@
+name: apply-code-format
+
+on: [workflow_dispatch]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: DoozyX/clang-format-lint-action@v0.13
+      with:
+        source: './src'
+        clangFormatVersion: 12
+        style: file
+        inplace: True
+    - uses: EndBug/add-and-commit@v4
```

Comment:
Maybe it is a bit safer to open a PR instead of directly pushing a new commit? There are Github actions for [automatically opening PRs](https://github.com/marketplace/actions/create-pull-request).